### PR TITLE
fix(i18n): remove old i18n configuration and add new workflow for intl

### PR DESCRIPTION
**Pull request by pavanjoshi914**
Describe the changes you have made in this PR -

* remove old i18n workflow
* add new i18n workflow
* translate circuitverse title for testing

### Before workflow

Current configuration of I18n uses a complex approach to feed translations in app and needs to be upgraded to a cleaner workflow.
  
Current workflow used requires CLI commands to generate files and needs to follow current guide

Create string resources for all the strings present in app
Create AppLocalizations class and AppLocalizationsDelegate
Generate .arb files from string resources 
Convert them into dart code by creating messages_all.dart 
Translations can be feed using AppLocalizations.of(context).title
 
### upgraded workflow

Upgraded workflow will provide easy steps to provide optimal I18n support in Flutter and considered to be used
 
Add intl package and flutter_localizations package with flag generate:true
Create l10n.yaml describing path of translation directory and name of localization class.
Create .arb files for providing translations.
Restart or reload app to trigger automated localization class generation.
use ```AppLocalizations.of(context).key_name```` to provide translation

### No need of explict list of supported locales

supported locales are automatically get defined in localization class when ```.arb``` file is created and reload is done, which can be accessed using ```AppLocalizations.of(context).SupportedLocales```

Screenshots of the changes (If any) -
image

image

Note: Please check Allow edits from maintainers. if you would like us to assist in the PR.

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -68,13 +68,11 @@ class CircuitVerseMobile extends StatelessWidget {
             builder: (themeContext) => GetMaterialApp(
               title: 'CircuitVerse Mobile',
               localizationsDelegates: [
```
Comment (manjotsidhu) [inline lib/main.dart:70]: If we are using generated `supportedLocales` then we should also be using generated delegates
```suggestion
              localizationsDelegates: AppLocalizations.localizationsDelegates,
```

File: lib/main.dart [hunk 2]
```diff
@@ -68,13 +68,11 @@ class CircuitVerseMobile extends StatelessWidget {
...
-                AppLocalizationsDelegate(),
+                AppLocalizations.delegate,
                 GlobalMaterialLocalizations.delegate,
                 GlobalWidgetsLocalizations.delegate,
               ],
```
Comment (manjotsidhu) [inline lib/main.dart:74]: If we are using generated `supportedLocales` then we should also be using generated delegates

localizationsDelegates: AppLocalizations.localizationsDelegates,

Comment (pavanjoshi914) [reply lib/main.dart]: using generated delegates instead of defining them explicitly!

**Comment by pavanjoshi914** [comment]
@manjotsidhu tests fixed, have a look into this

**Review by Nitish145 — APPROVED** [review]

**Review by manjotsidhu — APPROVED** [review]
Tested, working as expected.

**Pull request merged**